saleor/checkout/utils.py

```diff
@@ -603,25 +604,41 @@ def get_valid_shipping_methods_for_checkout(
     )
 
 
-def is_valid_shipping_method(checkout_info: "CheckoutInfo"):
-    """Check if shipping method is valid and remove (if not)."""
-    if not checkout_info.shipping_method:
-        return False
-    if not checkout_info.shipping_address:
-        return False
+def get_valid_collection_points_for_checkout(
+    lines: Iterable["CheckoutLineInfo"],
+    checkout_info: "CheckoutInfo",
+    country_code: Optional[str] = None,
+    quantity_check: bool = True,
+):
+    """Return a collection of `Warehouse`s that can be used as a collection point.
 
-    valid_methods = checkout_info.valid_shipping_methods
-    if valid_methods is None or checkout_info.shipping_method not in valid_methods:
-        clear_shipping_method(checkout_info)
-        return False
-    return True
+    Note that `quantity_check=False` should be used, when stocks quantity will
+    be validated in further steps (checkout completion) in order to raise
+    'InsufficientProductStock' error instead of 'InvalidShippingError'.
+    """
+
+    if not is_shipping_required(lines):
+        return []
+    if not checkout_info.shipping_address:
+        return []
+    line_ids = [line_info.line.id for line_info in lines]
+    lines = CheckoutLine.objects.filter(id__in=line_ids)
+
+    return (
+        Warehouse.objects.applicable_for_click_and_collect(lines, country_code)
```

Comment:
Can't we have a single method `applicable_for_click_and_collect` which will accept optional flag which will determine if it should check stock quantity or not?

Reply:
I guess these queries are complex enough, and they not need another if/else clause.

Reply:
I've spent more time finding the difference between them than understanding what they're doing

Reply:
🤷‍♂️